### DISCUSSION TITLE
Running Airflow docker-compose With Google OAUTH Authentication

### DISCUSSION THREAD

**Discussion by asura-io** [apache/airflow#19800] [Q&A]
I'm trying to get Airflow to use Google OAuth instead of the standard username/password login. I'm using the official [docker-compose](https://github.com/apache/airflow/blob/main/docs/apache-airflow/start/docker-compose.yaml) file. 

Under `x-airflow-common`, I have the following environment variables set:

```
    AIRFLOW__WEBSERVER__AUTHENTICATE: 'True'
    AIRFLOW__WEBSERVER__AUTH_BACKEND: 'airflow.contrib.auth.backends.google_auth'
    AIRFLOW__GOOGLE__CLIENT_ID: 'my-client-id'
    AIRFLOW__GOOGLE__CLIENT_SECRET: 'my-client-secret'
    AIRFLOW__GOOGLE__OAUTH_CALLBACK_ROUTE: 'private-airflow-server-url/oauth2callback'
``` 

I'm extending the standard airflow image in a Dockerfile, so I have the following in my Dockerfile that I've placed right next to the `docker-compose.yaml` file. 

```
FROM apache/airflow:2.2.0-python3.8

RUN airflow db init
RUN airflow connections add spark_local --conn-host local[*] --conn-type Spark
RUN pip install 'apache-airflow[google_auth]'
```

I've verified that `apache-airflow[google_auth]` is installed on my `webserver` container, and the environment variables are set as expected. However, I'm still presented with the standard username/passward login option instead of a way to authenticate with Google OAUTH. 

If anyone could give me pointers and show me what I'm missing, that would be greatly appreciated!

**Comment by potiuk** [comment]
I think a very good guideline and step-by-step does not exist but you can see at earlier discussions https://github.com/apache/airflow/discussions/17471 and try to search google or stack overflow for someone who integrated Airflow with Google Auth. 

Also you might want to take a look at Flask Application Builder configuration of Oauth authentication - this is what Airflow uses under the hood:  https://flask-appbuilder.readthedocs.io/en/latest/security.html

And finally if you get it working - any contribution on our docs would be most welcome. I think we miss a good guide for that.

**Reply by jaimegm** [reply]
Hi @potiuk,
I recently Implemented google auth on my airflow Deployment. I would like to add a good guide for it but I don't know where to put it. Could you point me in the correct direction? Should I create a `.rst` file [here](https://github.com/apache/airflow/tree/main/docs/apache-airflow/security)?!

**Reply by potiuk** [reply]
This is better place: https://github.com/apache/airflow/blob/main/docs/apache-airflow/production-deployment.rst

**Reply by tshradheya** [reply]
@jaimegm could you share some details? Will be really helpful
